dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/logger/SdkInternalLogger.kt

```diff
@@ -253,6 +264,22 @@ internal class SdkInternalLogger(
         }
     }
 
+    private fun Map<String, Any?>?.putIfNotExistOrMapIsNull(key: String, value: Float?): MutableMap<String, Any?> {
+        val mutableMap = this?.toMutableMap() ?: mutableMapOf()
+        return mutableMap.apply { putIfNotExist(key, value ?: return@apply) }
```

Comment:
Made it even simpler, now there is only one method used. `toMap` is not quite required here IMO, because all other parameter types requires `Map` which implemented by MutableMap as well